frappe/public/js/frappe/form/controls/select.js

```diff
@@ -150,24 +152,30 @@ frappe.ui.form.add_options = function (input, options_list, sort) {
 	};
 })(jQuery);
 
-function parse_option(v) {
+function parse_option(v, dt, df) {
 	let value = null;
 	let label = null;
 	let is_disabled = false;
 	let is_selected = false;
 
 	if (!is_null(v)) {
+		let translation_context = dt
+			? frappe.scrub(dt) + (df ? " (Field: " + df + ")" : null)
```

Comment:
suggestion: https://github.com/frappe/frappe/pull/26425

Reply:
Interesting to see that we've working on the same feature without knowing each others.

I din't see there is already DocType contextualized translation base on DocType name

I've check your PR, and I've adapt mine, seems better to use df property than $(input).data('xxx').

I prefers my "new" implementation because with DocType and DocField there will be a real unicity of translation key

I wonder, in your PR, about 
` df.context `
? 
How do you set it, it's not code property of Docfield DocType.

Reply:
> Interesting to see that we've working on the same feature without knowing each others.

Actually it's [the code from Dokos](https://gitlab.com/dokos/dodock/-/blob/develop/frappe/public/js/frappe/form/controls/select.js), where it was fixed like 4 years ago, specifically because of the _Letter Head field align_ you mention in your PR :smile:. Your PR reminded me that it was already fixed there so I just took the opportunity to send our changes upstream to Frappe.

---

> `df.context`

It's useful when creating Dialogs, where there is no "doctype" so to speak:
const dialog = new frappe.ui.Dialog({
    title: "Test",
    fields: [
        {
            fieldtype: "Select",
            options: ["A", "B", "C"],
            context: "My Translation Context",  # here
            label: "Value",
            fieldname: "value",
        },
    ],
})
dialog.show()

https://frappeframework.com/docs/user/en/api/dialog

Reply:
Ok, good catch for df.context

I've just "reset" the PR to avoid multiples commits.

I keept your 
`this.df.context || this.df.parent || this.doctype`
because for the same non "doctyped" field  this.df.parent can also be catch in contextual translation